lib/cocoapods/installer/user_project_integrator.rb

```diff
@@ -112,8 +115,38 @@ def create_workspace
       # @return [void]
       #
       def integrate_user_targets
-        targets_to_integrate.sort_by(&:name).each do |target|
-          TargetIntegrator.new(target).integrate!
+        target_integrators = targets_to_integrate.sort_by(&:name).map do |target|
+          TargetIntegrator.new(target)
+        end
+
+        Config.instance.with_changes(:silent => true) do
+          deintegrator = Deintegrator.new
+          all_project_targets = target_integrators.map(&:user_project).flat_map(&:native_targets).uniq
+          all_native_targets = target_integrators.flat_map(&:native_targets).uniq
+          targets_to_deintegrate = all_project_targets - all_native_targets
```

Comment:
@segiddins I think this case does not handle the following podfile transition from :

```
pod 'MBProgressHUD'
```
to :

```
target 'test' do
	pod 'MBProgressHUD'
end
```

Because both two cases integrates with the user project target _`test`_, both `all_project_targets` and `all_native_targets` include `test` target. However, we should still deintegrate this target because in the first podfile `libPods.a` library had been linked to it, but in the second case `libPods-test.a` library will be linked, and we have to get rid of `libPods.a`.